test/unit/spec/userMenuSpec.js

```diff
@@ -0,0 +1,98 @@
+/* global describe, it, expect, jasmine, beforeEach, afterEach */
+
+define([
+    'jquery',
+    'userMenu',
+    'narrativeConfig',
+    'narrativeMocks'
+], ($, UserMenu, Config, Mocks) => {
+    'use strict';
+
+    const TEST_TOKEN = 'fake_token',
+        TEST_USER = 'test_user',
+        TEST_NAME = 'Test User',
+        TEST_EMAIL = 'test_user@kbase.us';
+
+    describe('Test the UserMenu module', () => {
+        beforeEach(() => {
+            jasmine.Ajax.install();
+            Mocks.mockJsonRpc1Call({
+                url: Config.url('user_profile'),
+                response: [{
+                    profile: {
+                        userdata: {
+                            gravatarDefault: 'identicon'
+                        }
+                    }
+                }]
+            });
+        });
+
+        afterEach(() => {
+            jasmine.Ajax.uninstall();
+        });
+
+        it('Should make a user menu object with expected elements', () => {
+            expect(UserMenu).toBeDefined();
+            expect(UserMenu.make).toBeDefined();
+            const $elem = $('<div>');
+            const userMenu = UserMenu.make({
+                target: $elem,
+                token: TEST_TOKEN,
+                userName: TEST_USER,
+                email: TEST_EMAIL,
+                displayName: TEST_NAME
+            });
+            ['start', 'render', 'logout'].forEach(fn => {
+                expect(userMenu[fn]).toBeDefined();
+            });
+        });
+
+        it('Should start and create a menu in the target node with an empty user profile', () => {
+            const $elem = $('<div>');
+            const userMenu = UserMenu.make({
+                target: $elem,
+                token: TEST_TOKEN,
+                userName: TEST_USER,
+                email: TEST_EMAIL,
+                displayName: TEST_NAME
+            });
```

Comment:
could have a little helper function to do this, since it's repeated three times

Reply:
Yup, done.